src/overlays/actors/ovl_Boss_07/z_boss_07.c

```diff
@@ -242,500 +637,7210 @@ ActorProfile Boss_07_Profile = {
+
+        ShrinkWindow_Letterbox_SetSizeTarget(27);
+        subCamAt.x = this->subCamAt.x;
+        subCamAt.y = this->subCamAt.y + subCamAtYOffset;
+        subCamAt.z = this->subCamAt.z;
+        Play_SetCameraAtEye(play, this->subCamId, &subCamAt, &this->subCamEye);
+    }
+}
+
+void Boss07_Mask_SetupDeathCutscene(Boss07* this, PlayState* play) {
+    this->actionFunc = Boss07_Mask_DeathCutscene;
+    Animation_MorphToLoop(&this->skelAnime, &gMajorasMaskJerkingAnim, -10.0f);
+    this->actor.world.rot.y = this->actor.shape.rot.y =
+        Math_Atan2F_XY(-this->actor.world.pos.z, -this->actor.world.pos.x) * (0x8000 / M_PIf);
+    this->cutsceneState = MAJORAS_MASK_DEATH_CS_STATE_STARTED;
+    this->cutsceneTimer = 0;
+    this->actor.flags &= ~ACTOR_FLAG_ATTENTION_ENABLED;
+    this->damagedTimer = 20;
+}
+
+void Boss07_Mask_DeathCutscene(Boss07* this, PlayState* play) {
+    Player* player = GET_PLAYER(play);
+    Vec3f subCamAtTarget;
+    Vec3f subCamEye;
+
+    this->cutsceneTimer++;
+    this->damagedTimer = 20;
+    this->maskEyeTexIndex = MAJORAS_MASK_EYE_DULL;
+    SkelAnime_Update(&this->skelAnime);
+    Boss07_SmoothStop(this, 0.5f);
+
+    Math_ApproachF(&this->actor.world.pos.x, 0.0f, 0.05f, 5.0f);
+    Math_ApproachF(&this->actor.world.pos.z, 0.0f, 0.05f, 5.0f);
+    Math_ApproachF(&this->actor.world.pos.y, 130.0f, 0.05f, 3.0f);
+
+    Matrix_Translate(this->actor.world.pos.x, this->actor.world.pos.y, this->actor.world.pos.z, MTXMODE_NEW);
+    Matrix_RotateYS(this->actor.shape.rot.y, MTXMODE_APPLY);
+
+    switch (this->cutsceneState) {
+        case MAJORAS_MASK_DEATH_CS_STATE_STARTED:
+            if (CutsceneManager_GetCurrentCsId() == CS_ID_NONE) {
+                Cutscene_StartManual(play, &play->csCtx);
+                Player_SetCsActionWithHaltedActors(play, &this->actor, PLAYER_CSACTION_1);
+                this->subCamId = Play_CreateSubCamera(play);
+                Play_ChangeCameraStatus(play, CAM_ID_MAIN, CAM_STATUS_WAIT);
+                Play_ChangeCameraStatus(play, this->subCamId, CAM_STATUS_ACTIVE);
+                this->cutsceneTimer = 0;
+                this->cutsceneState = MAJORAS_MASK_DEATH_CS_STATE_PLAYING;
+                this->subCamAtNext.z = 0.0f;
+                this->fireTimer = 120;
+                Play_EnableMotionBlur(150);
+
+                this->subCamEyeNext.x = 0.0f;
+                this->subCamEyeNext.y = -30.0f;
+                this->subCamEyeNext.z = 270.0f;
+
+                this->subCamAtNext.x = 0.0f;
+                this->subCamAtNext.y = -30.0f;
+
+                Matrix_MultVec3f(&this->subCamEyeNext, &this->subCamEye);
+                Matrix_MultVec3f(&this->subCamAtNext, &this->subCamAt);
+            } else {
+                break;
+            }
+            FALLTHROUGH;
+        case MAJORAS_MASK_DEATH_CS_STATE_PLAYING:
+            player->actor.world.pos.x = 0.0f;
+            player->actor.world.pos.z = BREG(87) + 250.0f;
+            player->actor.
```

Comment:
```suggestion
    Vec3f* velocity0 = velocity;
```
Or maybe `firstVelocity?`

Reply:
Is this comment attached to the wrong line of code? I think I know what you're trying to refer to, though. `vel0` was there when I got here, and I was a bit afraid to change it, lol. I can change it to `firstVelocity`, though